homeassistant/components/google_generative_ai_conversation/__init__.py

```diff
@@ -91,8 +95,40 @@ def append_files_to_prompt():
                     )
                     prompt_parts.append(uploaded_file)
 
+        async def wait_for_file_processing(uploaded_file: File) -> None:
+            """Wait for file processing to complete."""
+            while True:
+                LOGGER.debug(
+                    "Waiting for file `%s` to be processed, current state: %s",
+                    uploaded_file.name,
+                    uploaded_file.state,
+                )
+                uploaded_file = await client.aio.files.get(
+                    name=uploaded_file.name,
+                    config={"http_options": {"timeout": TIMEOUT_MILLIS}},
+                )
+                if uploaded_file.state not in (
+                    FileState.STATE_UNSPECIFIED,
+                    FileState.PROCESSING,
+                ):
+                    break
+                await asyncio.sleep(FILE_POLLING_INTERVAL_SECONDS)
+
+            if uploaded_file.state == FileState.FAILED:
```

Comment:
We could be here if PROCESSING or STATE_UNSPECIFIED. Shouldn't we raise an error?